docs/sphinx/source/contributing.rst

```diff
@@ -254,6 +254,39 @@ To build the docs locally, install the ``doc`` dependencies specified in the
 `setup.py <https://github.com/pvlib/pvlib-python/blob/master/setup.py>`_
 file. See :ref:`installation` instructions for more information.
 
+Example Gallery
+---------------
+
+The example gallery uses `sphinx-gallery <https://sphinx-gallery.github.io/>`_
+and is generated from script files in the
+`docs/examples <https://github.com/pvlib/pvlib-python/tree/master/docs/examples>`_
+directory.  sphinx-gallery will execute example files that start with
+``plot_`` and capture the output.
+
+Here is a starter template for new examples:
+
+.. code-block:: python
+
+    """
+    Page Title
+    ==========
+
+    A sentence describing the example.
+    """
+
+    # %%
+    # Explanatory text about the example, what it does,
+    # why it does it, etc.
```

Comment:
```suggestion
    # why it does it, etc.  Text before the first line of code `import pvlib` will be printed to the example's webpage.
```

Reply:
> Anyone know why the Edit on GitHub links don't work for the gallery examples?

I think what's happening is that `sphinx-gallery` generates intermediate `.rst` files from the example scripts and puts them in `sphinx/source/auto_examples`, making them look just like normal RST source files.  When sphinx itself runs, it doesn't realize those files were just auto-generated a moment ago and so it creates "edit on github" links assuming that those RST files exist on GH as well.

Looks like it isn't a trivial fix to make the links point to the original .py files -- MetPy handles it with a custom template: https://github.com/Unidata/MetPy/commit/8f1783af3bdb2b3d664faf6d3eeb8af9024fed85. If nobody is eager, I'll volunteer to try to get it working in a separate PR.